src/App.js

```diff
@@ -14,11 +17,99 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: [],
+  }
+
+  showGoodsList = () => {
+    this.setState({ goods: [...goodsFromServer] });
+  }
+
+  reverse = () => {
+    this.setState(state => ({
+      goods: [...state.goods].reverse(),
+    }));
+  }
+
+  sortAlphabetically = () => {
+    this.setState(state => ({
+      goods: [...state.goods].sort((a, b) => a.localeCompare(b)),
```

Comment:
Also, it would be great to move all the sortings logic inside the `GoodsList` component.
It's good to split the logic. `App` component should only have the button 'start' which will toggle the render of the `GoodsList`. There you should pass initial goods and sort them inside that component (not inside the `App`)